torch/csrc/autograd/utils/grad_layout_contract.h

```diff
@@ -39,6 +39,12 @@ inline bool obeys_layout_contract(
       }
     }
     return true;
+  } else if (variable.is_nested()) {
+    // TODO: Nested Tensor does not have an implementation of detach. The
+    // current implementation of nested tensor likely does obey the gradient
+    // contract and should return true, but this would likely change in the
+    // future
+    return false;
```

Comment:
My current understanding of why the planned change OK is that the concept of layout for nested tensors is the same as normal tensors, i.e. to check if two tensors have the same layout just check if the strides are the same. Is that correct?

Reply:
nit: maybe change
```
if (a && !b) {
} else if (b) {
} else {
}
```
to 
```
if (b) {
} else if (a) {
} else {
}
```